### PR TITLE
gui_file_dialog.h Fix Memory leaks

### PR DESCRIPTION
**Pull request by CrezyDud**
There was a lot of leaking

`SUMMARY: AddressSanitizer: 32360856 byte(s) leaked in 31603 allocation(s).`

(408 bytes of that was X11, so they don't count)

### CODE REVIEW

File: examples/custom_file_dialog/gui_file_dialog.h [hunk 1]
```diff
@@ -47,38 +47,38 @@
...
-    Vector2 size;
+	Vector2 position;
+	Vector2 size;
 
-    bool fileDialogActive;
+	bool fileDialogActive;
 
-    bool dirPathEditMode;
-    char dirPathText[256];
+	bool dirPathEditMode;
```
Comment (raysan5) [inline examples/custom_file_dialog/gui_file_dialog.h:55]: Please, use 4 spaces instead of TABS

Comment (CrezyDud) [reply examples/custom_file_dialog/gui_file_dialog.h]: Done

**Comment by raysan5** [comment]
Reviewed with a simpler solution

**Comment by CrezyDud** [comment]
@raysan5 your solution is 54 changes instead of 27

**Pull request closed without merging**